integrations/clickup/clickup/client.py

```diff
@@ -0,0 +1,172 @@
+from loguru import logger
+from typing import Any, AsyncGenerator
+
+from port_ocean.utils import http_async_client
+
+from .utils import timestamp_to_datetime
+
+
+WEBHOOK_NAME = "Port-Ocean-Events-Webhook"
+WEBHOOK_EVENTS = [
+    "listCreated",
+    "listUpdated",
+    "listDeleted",
+    "taskCreated",
+    "taskUpdated",
+    "taskDeleted",
+    "taskCommentPosted",
+    "taskCommentUpdated",
+]
+
+
+class ClickupClient:
+    def __init__(self, clickup_personal_token: str) -> None:
+        self.clickup_url = "https://api.clickup.com/api/v2"
+        self.clickup_personal_token = clickup_personal_token
+
+        self.api_auth_header = {"Authorization": self.clickup_personal_token}
+        self.client = http_async_client
+        self.client.headers.update(self.api_auth_header)
+
+    async def create_events_webhook(self, app_host: str) -> None:
+        teams = self.get_paginated_teams()
+        async for team_batch in teams:
+            for team in team_batch:
+                await self._create_team_events_webhook(app_host, team["id"])
+
+    async def _create_team_events_webhook(self, app_host: str, team_id: int) -> None:
+        webhook_target_app_host = f"{app_host}/integration/webhook"
+        webhooks_response = await self.client.get(
+            f"{self.clickup_url}/team/{team_id}/webhook"
+        )
+        webhooks_response.raise_for_status()
+        webhooks = webhooks_response.json()
+
+        existing_webhook = next(
+            (
+                webhook
+                for webhook in webhooks["webhooks"]
+                if webhook["endpoint"] == webhook_target_app_host
+            ),
+            None,
+        )
+
+        if existing_webhook:
+            logger.info(
+                f"Ocean real time reporting clickup webhook already exists [ID: {existing_webhook['id']}]"
+            )
+            return
+
+        webhook_create_response = await self.client.post(
+            f"{self.clickup_url}/team/{team_id}/webhook",
+            json={
+                "endpoint": webhook_target_app_host,
+                "events": WEBHOOK_EVENTS,
+            },
+        )
+        webhook_create_response.raise_for_status()
+        webhook_create = webhook_create_response.json()
+        logger.info(
+            f"Ocean real time reporting clickup webhook created "
+            f"[ID: {webhook_create['id']}, Team ID: {team_id}]"
+        )
+
+    async def get_paginated_teams(
+        self, params: dict[str, Any] = {}
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        teams_response = await self.client.get(
+            f"{self.clickup_url}/team", params=params
+        )
+        teams_response.raise_for_status()
+        teams = teams_response.json()["teams"]
+        yield teams
```

Comment:
I don't see where you are handling potential HTTPS errors such as 40X, 50X from the client. Please take a look at other integrations and see how we handle errors and apply it to all your functions.

Reply:
I thought that's what the `<Response>.raise_for_status()` is for. I saw it was used in both Linear and Jira

Reply:
That's fine then